### ISSUE TITLE
Issue with sending max tokens/tokens amount that has long decimals.

### ISSUE THREAD

**Issue by tmashuang** [MetaMask/metamask-extension#5682]
When viewing this token transfer tx https://etherscan.io/tx/0x5b3765cdf02648a42e0f307aa0b0d4a72d4151ecd13fd993f1decf5b2200ac35#decodetab, under the Input Data section, the value is 57888134750431890000. After reviewing  token value of the from address, https://etherscan.io/token/0xe41d2489571d322189246dafa5ebde1f4699f498?a=0x10ac56c7adac67d1a50002540c7a000bb5b12530, seems we are rounding up at the end.

TLDR; 
Input Value:    57888134750431890000
Actual Value: 57888134750431889787

**Comment by danfinlay** [comment]
Probably the same as / Related to #5180
